internal/app/cloudinfo/loader/manager.go

```diff
@@ -0,0 +1,110 @@
+// Copyright © 2019 Banzai Cloud
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package loader
+
+import (
+	"context"
+	evbus "github.com/asaskevich/EventBus"
+	"github.com/banzaicloud/cloudinfo/pkg/cloudinfo"
+	"github.com/goph/emperror"
+	"github.com/goph/logur"
+	"github.com/spf13/viper"
+)
+
+// ServiceManager abstracts the operations related to cloud info services
+type ServiceManager interface {
+	// ConfigureServices parses the service configuration file and registers supported services
+	ConfigureServices(ctx context.Context, providers []string)
+
+	// LoadServiceInformation triggers importing cloud information based on the available service information
+	LoadServiceInformation(ctx context.Context, providers []string)
+}
+
+// defaultServiceManager default implementation for the service manager
+type defaultServiceManager struct {
+	// (all) services parsed from the configuration
+	services map[string][]ServiceData
+
+	// the destination of the loaded data
+	store cloudinfo.CloudInfoStore
+
+	// component logger instance
+	log logur.Logger
+
+	// component eventbus instance
+	bus evbus.Bus
+}
+
+func (sm *defaultServiceManager) LoadServiceInformation(ctx context.Context, providers []string) {
+	sm.log.Info("triggering cloud information importing ...")
+	for _, provider := range providers {
+
+		for _, service := range sm.services[provider] {
+			if !service.IsStatic {
+				sm.log.Debug("skip loading static cloud information, dynamic service", map[string]interface{}{"service": service.Name})
+				continue
+			}
+
+			NewLoaderEvents(sm.bus).NotifyScrapeCompleted(
```

Comment:
Signal the completion of data scraping with different topics per cloud provider. This will allow better separation of refreshing impacted caches.